.github/workflows/apk.yml

```diff
@@ -14,23 +14,42 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v3
-
-    - name: Set up JDK
-      uses: actions/setup-java@v3
-      with:
-        distribution: 'zulu'  # Use Zulu OpenJDK
-        java-version: '11'
-
-    - name: Install dependencies
-      run: ./gradlew dependencies
-
-    - name: Build APK
-      run: ./gradlew assembleRelease
-
-    - name: Upload APK
-      uses: actions/upload-artifact@v3
-      with:
-        name: "${{ github.event.inputs.name }}-APK"
-        path: app/build/outputs/apk/release/app-release.apk
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+        # Load google-services.json and local.properties from the secrets
+      - name: Decode secrets
+        env:
+          GOOGLE_SERVICES: ${{ secrets.GOOGLE_SERVICES }}
+          # LOCAL_PROPERTIES: ${{ secrets.LOCAL_PROPERTIES }}
+        run: |
+          echo "$GOOGLE_SERVICES" | base64 --decode > ./app/google-services.json
+
+      - name: Set permissions for google-services.json
+        run: chmod 644 ./app/google-services.json
+
+      - name: Set executable permissions for gradlew
+        run: chmod +x ./gradlew
```

Comment:
I think that you can replace `chmod +x ./gradlew` by `chmod u+x ./gradlew` this gives execution permission only to the owner and not to all users.